vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestRunListener.kt

```diff
@@ -62,43 +67,53 @@ class ScreenRecorderTestRunListener(
     }
 
     override suspend fun testAssumptionFailure(test: TestIdentifier, trace: String) {
-        pullVideo(test.toTest())
+        //Treating as failure
+        hasFailed = true
     }
 
     override suspend fun testEnded(test: TestIdentifier, testMetrics: Map<String, String>) {
         pullVideo(test.toTest())
+        lastRemoteFile = null
     }
 
     override suspend fun testRunFailed(errorMessage: String) {
```

Comment:
Is this fun called after needed attempts were executed?

Reply:
It depends. The lifecycle of am instrument output sometimes doesn't send testEnded event in edge-cases, but this method is called after each attempt by default regardless of the test result. If no other methods are reported, then the test is considered passed